### PR TITLE
Factor out some of the code shared between sytest scripts

### PR DESCRIPTION
**Pull request by NegativeMjark**

### CODE REVIEW

File: run-tests.pl [hunk 1]
```diff
@@ -94,7 +94,7 @@
 
    'bind-host=s' => \$BIND_HOST,
 
-   'p|port-range=s' => \(my $PORT_RANGE = "8800:8819"),
+   'p|port-range=s' => \(my $PORT_RANGE = "8800:8900"),
```
Comment (richvdh) [inline run-tests.pl]: `port-max` appears to be inclusive so this might be better as `8899` ?

File: jenkins/install_and_run.sh [hunk 2]
```diff
@@ -14,6 +14,21 @@ cd "`dirname $0`/.."
...
+: ${BIND_HOST=localhost}
+
+export PORT_BASE
+export PORT_COUNT
+export BIND_HOST
+
 ./jenkins/kill_old_listeners.sh
 
-./run-tests.pl --bind-host ${BIND_HOST:-localhost} -O tap --all "$@" > results.tap
+export TOX_BIN=$WORKSPACE/.tox/py27/bin
```
Comment (richvdh) [inline jenkins/install_and_run.sh]: is there a reason to prefer `$WORKSPACE` over `pwd` here? seems like this unnecessarily ties it to jenkins.

Comment (NegativeMjark) [inline jenkins/install_and_run.sh]: Yes, using WORKSPACE means that this script will also work when sytest is checked out as a subfolder in the synapse and dendron jenkins tests.

File: run-tests.pl (hunk 1 above)
Comment (NegativeMjark) [inline run-tests.pl]: Done 3ea0c89

File: jenkins/clone.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
+#! /bin/bash
+
+NAME=$1
+PROJECT=$2
+BASE=".$NAME-base"
+
+# update our clone
+if [ ! -d .$NAME-base ]; then
```
Comment (richvdh) [inline jenkins/clone.sh]: this script could do with some quotes around its $vars, so as to reduce breakage in the event of spaces in filenames

File: jenkins/clone.sh [hunk 4]
```diff
@@ -0,0 +1,24 @@
+#! /bin/bash
```
Comment (richvdh) [inline jenkins/clone.sh]: could do with a comment saying what this does and what the args are

File: jenkins/clone.sh [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+git clone $BASE $NAME --shared
+
+: ${GIT_BRANCH:="origin/$(git rev-parse --abbrev-ref HEAD)"}
+cd $NAME
+# check out the relevant branch
+git checkout "${GIT_BRANCH}" || (
+    echo >&2 "No ref ${GIT_BRANCH} found, falling back to develop"
+    git checkout "origin/develop"
+)
+git clean -df .
```
Comment (richvdh) [inline jenkins/clone.sh]: this doesn't feel like it belongs here

**Comment by NegativeMjark** [comment]
@richvdh PTAL

File: jenkins/install_and_run.sh (hunk 2 above)
Comment (richvdh) [inline jenkins/install_and_run.sh]: ugh. fair enough

**Comment by richvdh** [comment]
lgtm

**Pull request merged**